### PR TITLE
Cleanup IndexerRegistry contracts

### PR DESCRIPTION
**Pull request by nz-neooo**

### CODE REVIEW

File: contracts/IndexerRegistry.sol [hunk 1]
```diff
@@ -59,30 +59,35 @@ contract IndexerRegistry is Initializable, OwnableUpgradeable, Constants {
...
-    /// @notice Indexer controller account => main account
-    mapping(address => address) public controllerToIndexer;
-    /// @notice The minimum stake amount for Indexer, set by owner.
+
+    /// @notice The minimum stake amount for Indexer, set by owner
     uint256 public minimumStakingAmount;
-    // Delegation tax rate per indexer
+
+    /// @notice Indexer's metadata: indexer => metadata, if metadata = 0, no indexer
+    mapping(address => bytes32) public indexers;
```
Comment (mzxyz) [inline contracts/IndexerRegistry.sol]: [suggestion] Better name: `metadata` or `indexerMetadata`, similar with `controllers`

File: contracts/IndexerRegistry.sol [hunk 2]
```diff
@@ -161,52 +158,57 @@ contract IndexerRegistry is Initializable, OwnableUpgradeable, Constants {
...
-        // remove previous controller to indexer link
-        address prevController = indexerToController[msg.sender];
-        delete controllerToIndexer[prevController];
-
-        // add 2 directional links between indexer and controller
-        indexerToController[msg.sender] = _controller;
-        controllerToIndexer[_controller] = msg.sender;
-        emit SetControllerAccount(msg.sender, _controller);
+    function setControllerAccount(address controller) external {
+        require(indexers[msg.sender] != bytes32(0), 'G002');
```
Comment (mzxyz) [inline contracts/IndexerRegistry.sol]: [suggestion] use internal function like `isIndexer` or modifier to replace the duplicate check `indexers[msg.sender] != bytes32(0)`

**Comment by mzxyz** [comment]
Awesome work 👍 the code is much clean now.
Left 2 small suggestions, LGTM

File: contracts/IndexerRegistry.sol (hunk 2 above)
Comment (nz-neooo) [reply contracts/IndexerRegistry.sol]: Used modifier

File: contracts/IndexerRegistry.sol (hunk 1 above)
Comment (nz-neooo) [reply contracts/IndexerRegistry.sol]: fixed

**Review by mzxyz — APPROVED** [review]

**Pull request merged**